pkg/config/content.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+// ContentV1 is the v1 configuration for a generic remote content provider.
+type ContentV1 struct {
+	Source   string             `json:"source"`
+	Timeouts *ContentV1Timeouts `json:"timeouts,omitempty"`
+	//TODO(lucab): specify this better
+	CertificateAuthorities []ContentV1CertAuth `json:"certificateAuthorities,omitempty"`
+}
+
+// ContentV1Timeouts records HTTPS client timeouts
+type ContentV1Timeouts struct {
+	HTTPResponseHeaders int `json:"httpResponseHeaders"`
+	HTTPTotal           int `json:"httpTotal"`
+}
+
+// ContentV1CertAuth records HTTPS client custom CAs
+type ContentV1CertAuth struct {
+	//TODO(lucab): store PEM here? Or just a path?
```

Comment:
@dgonyeo this is the discussion we started on Ignition PR. There are a few options here:
 1. store the PEMs inline. No need for separate disk serialization from Ignition. But this add data duplication proportional to the number of volumes/keyslots. Also, cert rotation will be less than ideal.
 1. store certs somewhere in `/boot/etc/` and just record the path here. I'm unsure about naming scheme for the file on disk.
 1. add a `name` field to each `certificateAuthorities` entry, and use it here as an index. It needs a schema change before we stabilize 2.2.0, and some pre-agreed path (e.g. `/boot/.../$name.crt`).
 1. don't keep any specific CA reference here, and just source all PEMs available in some specific directory. This means we don't want to pin specific CAs to specific endpoint, now or in the future.

Reply:
I'm tempted to go with option 2. For the naming scheme here's a couple ideas:
1. we could translate the source of the cert into something that's valid as a filename and use that (although dataurl's would probably need to be handled differently due to length)
1. we could try to infer the name based on the URL (ex: `https://example.com/foo/bar.pem` would be named `bar.pem`)
1. name them based on their index in the Ignition config (ex: `0.pem`, `1.pem`, `2.pem`, ...)
1. we could go with something content-addressable, and name all certs based on a hash of the cert

Reply:
I would either go for the array index in the ignition config (to ease manual certificate rotation), or with `c_rehash` style of hashing. Slight preference for the former, as I currently don't see any benefit/interop we gain from with latter.

/cc @bgilbert @ajeddeloh as this was part of yesterday discussion